src/PostgREST/ApiRequest/Preferences.hs

```diff
@@ -110,34 +116,39 @@ data Preferences
 --     , preferTransaction = Just Commit
 --     , preferMissing = Just ApplyDefaults
 --     , preferHandling = Just Strict
+--     , preferTimezone = Nothing
 --     , invalidPrefs = [ "anything" ]
 --     }
 --
-fromHeaders :: Bool -> [HTTP.Header] -> Preferences
-fromHeaders allowTxEndOverride headers =
+fromHeaders :: Bool -> TimezoneNames -> [HTTP.Header] -> Preferences
+fromHeaders allowTxDbOverride tzNames headers =
   Preferences
     { preferResolution     = parsePrefs [MergeDuplicates, IgnoreDuplicates]
     , preferRepresentation = parsePrefs [Full, None, HeadersOnly]
     , preferParameters     = parsePrefs [SingleObject]
     , preferCount          = parsePrefs [ExactCount, PlannedCount, EstimatedCount]
-    , preferTransaction    = if allowTxEndOverride then parsePrefs [Commit, Rollback] else Nothing
+    , preferTransaction    = if allowTxDbOverride then parsePrefs [Commit, Rollback] else Nothing
     , preferMissing        = parsePrefs [ApplyDefaults, ApplyNulls]
     , preferHandling       = parsePrefs [Strict, Lenient]
+    , preferTimezone       = getTimezoneFromPrefs
     , invalidPrefs         = filter (`notElem` acceptedPrefs) prefs
     }
   where
     mapToHeadVal :: ToHeaderValue a => [a] -> [ByteString]
     mapToHeadVal = map toHeaderValue
+    timezonePrefs = map ((<>) "timezone=" . encodeUtf8) tzNames
```

Comment:
@taimoorzaeem Hm, this looks inefficient. There's like 1K timezones and we're mapping over them every time right? I assume this should be avoidable as we only need a search?

Reply:
Avoiding the encoding `encodeUtf8` would be good for perf too. I've suggested to convert `TimezoneNames` to a `Set ByteString` below.